### PR TITLE
Implementation of Lane and LaneSet. Resolves #53

### LINKED ISSUE #53: Implement the Lane and LaneSet
As a process designer I would like to add lanes to my diagram

### PR DESCRIPTION
**Pull request by caleeli**
This PR implements the bpmn elements: Lane and LaneSet, whit which can create processes like this:

image

### CODE REVIEW

File: src/ProcessMaker/Nayra/Bpmn/Lane.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<?php
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Bpmn/Lane.php]: Implementation of the Lane class

File: src/ProcessMaker/Nayra/Bpmn/LaneSet.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
+<?php
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Bpmn/LaneSet.php]: Implementation of the LaneSet class and its properties.

File: src/ProcessMaker/Nayra/Bpmn/ProcessTrait.php [hunk 3]
```diff
@@ -53,6 +53,7 @@
     protected function initProcessTrait()
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Bpmn/ProcessTrait.php]: This change adds LaneSets to a process

File: src/ProcessMaker/Nayra/Contracts/Bpmn/LaneInterface.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<?php
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Contracts/Bpmn/LaneInterface.php]: This file declares the Lane interface.

File: src/ProcessMaker/Nayra/Contracts/Bpmn/LaneSetInterface.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<?php
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Contracts/Bpmn/LaneSetInterface.php]: This file declares the LaneSet interface.

File: src/ProcessMaker/Nayra/Contracts/Bpmn/ProcessInterface.php [hunk 6]
```diff
@@ -26,6 +26,7 @@
     const BPMN_PROPERTY_IS_EXECUTABLE = 'isExecutable';
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Contracts/Bpmn/ProcessInterface.php]: This change declares the lane set property for a process

File: tests/Feature/Engine/LanesTest.php [hunk 7]
```diff
@@ -0,0 +1,140 @@
+<?php
```
Comment (caleeli) [inline tests/Feature/Engine/LanesTest.php]: This file tests the implementation of Lanes and Lane Sets

File: tests/ProcessMaker/Models/ProcessRepository.php [hunk 8]
```diff
@@ -2,6 +2,8 @@
```
Comment (caleeli) [inline tests/ProcessMaker/Models/ProcessRepository.php]: This change add a builder for the Lanes and the LaneSets

File: tests/ProcessMaker/Repositories/BpmnFileElement.php [hunk 9]
```diff
@@ -18,8 +18,11 @@
 use ProcessMaker\Nayra\Contracts\Bpmn\FlowNodeInterface;
```
Comment (caleeli) [inline tests/ProcessMaker/Repositories/BpmnFileElement.php]: This change enable to load lanes and laneSets from a bpmn file.

File: src/ProcessMaker/Nayra/Bpmn/LaneSet.php [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+     */
+    public function setName($name)
+    {
+        return $this->setProperty(self::BPMN_PROPERTY_NAME, $name);
+    }
+
+    /**
+     * Set the lanes of the lane set.
+     *
+     * @param \ProcessMaker\Nayra\Contracts\Bpmn\CollectionInterface $lanes
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/LaneSet.php]: just CollectionInterface is enough in the documentation comment

File: src/ProcessMaker/Nayra/Bpmn/ProcessTrait.php [hunk 11]
```diff
@@ -323,4 +324,29 @@ public function call(DataStoreInterface $dataStore = null)
...
\ No newline at end of file
+
+    /**
+     * Set the lane sets of the process
+     *
+     * @param CollectionInterface $laneSets
+     *
+     * @return $this
+     */
+    public function getLaneSets()
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/ProcessTrait.php]: Do this returns $this? it seems a collection

File: src/ProcessMaker/Nayra/Bpmn/Lane.php [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    public function setFlowNodes(CollectionInterface $nodes)
+    {
+        return $this->setProperty(self::BPMN_PROPERTY_FLOW_NODE, $nodes);
+    }
+
+    /**
+     * Set the child lanes of the lane.
+     *
+     * @param \ProcessMaker\Nayra\Contracts\Bpmn\CollectionInterface $lanes
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/Lane.php]: Justo Collection interface in the doc.

File: src/ProcessMaker/Nayra/Bpmn/LaneSet.php [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+     */
+    public function getName()
+    {
+        return $this->getProperty(self::BPMN_PROPERTY_NAME);
+    }
+
+    /**
+     * Get the lanes of the lane set.
+     *
+     * @return \ProcessMaker\Nayra\Contracts\Bpmn\CollectionInterface
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/LaneSet.php]: just CollectionInterface id the comment

File: src/ProcessMaker/Nayra/Bpmn/ProcessTrait.php (hunk 11 above)
Comment (caleeli) [reply src/ProcessMaker/Nayra/Bpmn/ProcessTrait.php]: Fixed. It returns a Collection of LaneSetInterface

File: src/ProcessMaker/Nayra/Bpmn/Lane.php (hunk 12 above)
Comment (caleeli) [reply src/ProcessMaker/Nayra/Bpmn/Lane.php]: It is better to use the complete name of a class in the documentation blocks.

**Pull request merged**